src/modules/RA/DataLayer/components/tabs/AddField.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useCallback } from 'react';
+
+import Button from '@material-ui/core/Button';
+import Modal from '@material-ui/core/Modal';
+import Select from '@material-ui/core/Select';
+import  AddIcon from '@material-ui/icons/Add';
+import CancelIcon from '@material-ui/icons/Cancel';
+
+import { useTranslate } from 'react-admin';
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles({
+  modal: {
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    position: 'absolute',
+    width: 500,
+    backgroundColor: 'white',
+    padding: '1em',
+  },
+  modalButtons: {
+    marginTop: '1em',
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  fieldSelect: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+});
+
+const AddPopupField = ({ fields, onAdd, textContent }) => {
+  const classes = useStyles();
+  const translate = useTranslate();
+
+  const [showModal, setShowModal] = useState(false);
+  const [selected, setSelected] = useState('');
+
+  const onSelect = useCallback(e => {
+    setSelected(parseInt(e.target.value, 10));
+  }, []);
+
+  const onClose = useCallback((addField = false) => {
+    if (addField) {
+      onAdd(selected);
+    }
+    setSelected('');
+    setShowModal(false);
+  }, [selected, onAdd]);
+
+  const showModalOnclick = useCallback(() => setShowModal(true), []);
+
+  return (
+    <>
+      <Button
+        type="button"
+        onClick={showModalOnclick}
+        label="datalayer.form.popup.add-field"
+        variant="contained"
+        startIcon={<AddIcon />}
+      >
+        {translate(textContent.addField)}
+      </Button>
+      <Modal open={showModal} onClose={onClose}>
+        <div className={classes.modal}>
+          <div className={classes.fieldSelect}>
+            <Select native value={selected} onChange={onSelect}>
+              <option value="">{translate(textContent.selectField)}</option>
+              {fields.map(field => (
+                <option
+                  key={field.sourceFieldId}
+                  value={field.sourceFieldId}
+                >
+                  {field.label} ({field.name})
+                </option>
+              ))}
+            </Select>
+          </div>
+
+          <div className={classes.modalButtons}>
+            <Button
+              variant="outlined"
+              color="secondary"
+              onClick={() => onClose(false)}
+              startIcon={<CancelIcon />}
+            >
+              {translate(textContent.cancel)}
+            </Button>
+            <Button
+              variant="outlined"
+              color="primary"
+              onClick={() => onClose(true)}
+              startIcon={<AddIcon />}
+              disabled={!selected}
+            >
+              {translate(textContent.add)}
+            </Button>
+          </div>
+        </div>
```

Comment:
Maybe you should put this whole code into a `<form>` tag and manage it in uncontrolled way?
It would clean your code and it's semantically better.

```jsx 
<form onSubmit={handleSubmit}>
          <Select name="selected" native value={selected} onChange={onSelect}>
              <option value="">{translate(textContent.selectField)}</option>
              {fields.map(field => (
                <option
                  key={field.sourceFieldId}
                  value={field.sourceFieldId}
                >
                  {field.label} ({field.name})
                </option>
              ))}
          </Select>
    <Button type="cancel"> // It will reset your form
    <Button type="submit"> // it will submit your form
</form>
```

and 

```jsx
  const handleSubmit = event => {
    event.preventDefault();
    const formData = new FormData(event.target);
    const selected = formData.get('selected');
    onAdd(Number(selected));
    setShowModal(false);
  };
```

And you can rid off `onSelect` handler and `selected` useState.
To be tested

Reply:
Done.